en/components/grids_templates/paging.md

```diff
@@ -55,97 +55,141 @@ The following example represents @@igComponent pagination and exposes the option
 
 ```html
 <ng-template #myTemplate let-grid>
-    Current page: {{ @@igObjectRef.page }}
-    <input type="number" [(ngModel)]="grid.page" />
-    Total pages: {{ @@igObjectRef.totalPages }}
+    Total records: {{ this.totalRecords }}
+    Records per page: {{ this.perPage }}
+    Current page: {{ this.page }}
+    <input type="number" [(ngModel)]="this.page" (change)="valueChange(this.page)" />
+    Total pages: {{ this.totalPages }}
 </ng-template>
 
 <@@igSelector [paging]="true" [paginationTemplate]="myTemplate">
     ...
 </@@igSelector>
 ```
 
-Paging can also be done programmatically through the @@igComponent API, using the [`paginate`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#paginate), [`previousPage`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#previouspage), [`nextPage`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#nextpage) methods and the inputs [`page`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#page), [`perPage`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#perpage) and [`totalRecords`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#totalrecords). Where *page* allows you to set the current page, *perPage* - the number of items that are displayed at one page and *totalRecords* - the number of the records that are in the grid. `TotalRecords` property is useful when you have paging with remote data and you want to alter the page count based on total remote records. Keep in mind that If you are using paging and all the data is passed to the grid, the value of totalRecords property will be set by default to the length of the provided data source. If totalRecords is set, it will take precedence over the default length based on the data source."
-
 ```typescript
-// Go to page 6
-this.@@igObjectRef.paginate(5);
-
-// Go to previous/next page
-this.@@igObjectRef.previousPage();
-this.@@igObjectRef.nextPage();
-
-// Check for first/last page
-this.@@igObjectRef.isFirstPage;
-this.@@igObjectRef.isLastPage;
+public totalRecords = 0;
+public totalPages = 0;
+public page = 0;
+public perPage = 10;
+
+public ngOnInit() {
+    this.totalRecords = allData.length;
+    this.totalPages = Math.ceil(this.totalRecords / this.perPage);
+    this.data = this.sliceData(this.page, this.perPage);
+}
 
-// Get the number of pages
-this.@@igObjectRef.totalPages;
+public valueChange(newPage: number) {
+    this.data = this.sliceData(newPage, this.perPage);
+}
 
-// Change the number of records per page
-this.@@igObjectRef.perPage = 25;
+public sliceData(page, perPage) {
+    const start = page * perPage;
+    const end = (page + 1) * perPage;
+    const result = localData.slice(start, end);
+    return result;
+}
+```
 
-// Enables/disables paging
-this.@@igObjectRef.paging = false;
+or pipe the `data` input value through a custom Angular pipe - this way a developer need not worry about change detection in certain scenarios, Angular will handle it internally:
```

Comment:
Let's start with uppercase, or something is missing from the beginning of the sentence?